### PR TITLE
recipe(owlvit): add xenova owlvit cpu recipe

### PR DESCRIPTION
**Pull request by fuliucansheng**
## Summary

Adds a verified CPU fp32 direct-ONNX `zero-shot-object-detection` recipe for `Xenova/owlvit-base-patch32`. The Xenova checkpoint publishes ONNX artifacts for Transformers.js but does not ship PyTorch/safetensors weights, so this recipe targets the Hub-hosted `onnx/model.onnx` artifact with `export: null`.

This refresh fixes the recipe itself by adding the missing `loader.task` declaration, rebases the branch onto current `origin/main`, and keeps only the exact CPU fp32 tuple that builds and runs perf successfully. A CPU fp16 candidate was re-tested from `onnx/model_fp16.onnx`; it preserves FLOAT16 at build time but fails CPU perf/session initialization, so it is not shipped as a CPU recipe.

## Model metadata

**What the model does:** `Xenova/owlvit-base-patch32` is an OWL-ViT zero-shot object detection model. It consumes tokenized text queries plus an RGB image tensor and emits class-query logits, predicted boxes, text embeddings, and image embeddings.

**Primary user stories:**

| Story | Confidence |
|---|---|
| A user supplies an image and text prompts to obtain zero-shot object detection scores and boxes for open-vocabulary localization. | verified from checkpoint task metadata, config, and ONNX graph I/O |
| A developer consumes the Xenova ONNX artifact directly and runs WinML CPU inference without PyTorch checkpoint weights. | verified from repository file layout and direct-ONNX build path |

**Supported tasks:**

| Task | Surface | Confidence |
|---|---|---|
| `zero-shot-object-detection` | checkpoint metadata, Transformers/Optimum task resolution, WinML direct-ONNX recipe | verified for CPU fp32 build/perf |

**Model architecture:**

```text
OwlViTForObjectDetection
|-- Vision tower (ViT base patch32, image size 768)
|   |-- Patch embedding
|   |-- Transformer encoder stack x12
|   `-- Image embedding grid [24, 24, 768]
|-- Text tower (CLIP-style text encoder, max positions 16)
|   |-- Token embeddings (vocab 49408)
|   `-- Transformer encoder stack x12
`-- Detection heads
    |-- Class-query logits [image_batch, 576, num_queries]
    `-- Box prediction head [image_batch, 576, 4]
```

Source/confidence: pinned checkpoint config resolves `model_type: owlvit`, `architectures: [OwlViTForObjectDetection]`, image size 768, patch size 32, text max positions 16, and hidden sizes 768/512; graph I/O verifies the direct ONNX contract.

### Baseline

Baseline was refreshed from current `origin/main` at `e405dc6638dff6cac96b1539ee0cea678cc6d183` with `winml, version 0.2.0`.

| Probe | Result |
|---|---|
| `winml inspect -m Xenova/owlvit-base-patch32` | PASS: resolves task `zero-shot-object-detection`, loader `AutoModelForZeroShotObjectDetection`, model type `owlvit`, exporter `OwlViTOnnxConfig`, and generic WinML inference. |
| Recipe-free HF build | FAIL: `Xenova/owlvit-base-patch32` does not publish PyTorch, safetensors, TensorFlow, Flax, or checkpoint weights, so the HF export path cannot build from the model ID. |
| Direct-ONNX CPU auto-config | PASS: `winml config -m Xenova/owlvit-base-patch32/onnx/model.onnx --ep cpu --device cpu` emits `export: null`, `optim: {}`, `quant: null`, and `compile: null`, but no task metadata. |
| Direct ONNX graph | PASS: `onnx/model.onnx` is opset 14 with inputs `input_ids`, `pixel_values`, `attention_mask`; outputs `logits`, `pred_boxes`, `text_embeds`, `image_embeds`; initializers are FLOAT. |
| L3 task metric | CLI-BLOCKED: `winml eval --schema --task zero-shot-object-detection` reports the task is not supported by the eval registry. |

### Goal

Effort: L0. Goal ceiling: L1 on reachable CPU fp32. Outcome: L0 recipe-only.

Success means the checked-in CPU fp32 direct-ONNX recipe builds a structurally valid ONNX artifact and runs `winml perf` on CPU with explicit dynamic-shape overrides for the model's 768 x 768 image contract. CPU fp16 was attempted as a candidate but is not part of the shipped recipe set because L1 fails on this host.

### Outcome

Highest verified goal: L1 PASS on CPU fp32. Coverage is full for the shipped tuple (`cpu/cpu/fp32`) and no accelerator tuples are claimed.

Shipped recipe:

| Recipe |
|---|
| `examples/recipes/Xenova_owlvit-base-patch32/cpu/cpu/zero-shot-object-detection_fp32_config.json` |

No methodology friction observed in this rerun. The direct-ONNX dynamic shape requirement is captured as model-specific reproduction evidence: perf needs `--shape-config` so random inputs use `pixel_values [1,3,768,768]` and token tensors `[1,16]`.

### Per-EP/device/precision results

| Tier | EP / Device | Precision | Verdict | Mean | p50 | Throughput | RAM delta | Task metric |
|---|---|---:|---|---:|---:|---:|---:|---|
| L0 build | CPUExecutionProvider / cpu | fp32 | PASS | - | - | - | - | - |
| L1 perf | CPUExecutionProvider / cpu | fp32 | PASS | 137.267 ms | 131.513 ms | 7.29 samples/s | +611.49 MB total | - |
| L0 candidate | CPUExecutionProvider / cpu | fp16 | PASS, not shipped | - | - | - | - | - |
| L1 candidate | CPUExecutionProvider / cpu | fp16 | FAIL, not shipped | - | - | - | - | ORT session initialization fails on CPU for the fp16 graph. |
| L3 eval | CPUExecutionProvider / cpu | fp32 | CLI-BLOCKED | - | - | - | - | `zero-shot-object-detection` is not supported by `winml eval`. |

CPU fp32 structural validation: ONNX IR 7, opset 14, inputs `input_ids [text_batch_size, sequence_length] int64`, `pixel_values [image_batch_size, num_channels, height, width] float32`, and `attention_mask [text_batch_size, sequence_length] int64`; outputs `logits [image_batch_size, 576, num_queries]`, `pred_boxes [image_batch_size, 576, 4]`, `text_embeds [text_batch_size, max_text_queries, 512]`, and `image_embeds [image_batch_size, 24, 24, 768]`.

CPU fp16 candidate evidence: direct-ONNX build from `Xenova/owlvit-base-patch32/onnx/model_fp16.onnx` completed in 1.5s and preserved 410 FLOAT16 initializers, but `winml perf` failed during CPU session initialization with an ORT graph error referencing `InsertedPrecisionFreeCast_...SimplifiedLayerNormFusion`. The fp16 candidate recipe is omitted from the final diff.

Provider snapshot during validation: `['AzureExecutionProvider', 'CPUExecutionProvider']`.

### Delta

Recipe-vs-direct-ONNX CPU auto-config changes:

| JSON pointer | Baseline value | Shipped value | Reason |
|---|---|---|---|
| `/export` | `null` | `null` | Direct ONNX artifact; no PyTorch export. |
| `/quant` | `null` | `null` | CPU fp32 recipe, no quantization. |
| `/loader/task` | absent | `zero-shot-object-detection` | Make the checked-in recipe self-describing for the model task instead of a taskless direct-ONNX shell. |

No source code or tests are changed, and `examples/recipes/README.md` is untouched.

Reducibility: recipe-only remains appropriate because the broken HF build path is checkpoint packaging-specific: the Xenova repo publishes ONNX artifacts but no PyTorch weights. The recipe declares this direct-artifact intent; no safe class-wide source change is required for all OWL-ViT checkpoints.

### Analyze summary

Static rule analysis produced parseable JSON but exited nonzero because dynamic-shape operators lacked sufficient rule information and OpenVINO reported unsupported operators. This is compatibility analysis, not runtime execution.

**Component-level summary**

| Artifact | Architecture coverage | Mapping | Actionable EP findings |
|---|---|---|---|
| fp32 | vision patch embedding/encoder; text encoder; detection heads | 1923 optimized ONNX ops analyzed; ONNX graph scopes identify OWL-ViT vision/text/head regions | QNN/OpenVINO/VitisAI all flag dynamic input parsing limitations for NPU analysis. OpenVINO reports unsupported `Mul`, `Sub`, `Add`, `Div`, `Cast`, and `Reciprocal`; QNN has many dynamic-shape unknowns including `Expand`, `Gather`, `Shape`, `Range`, `Einsum`, and `Erf`. |

**Op-level summary**

| Artifact | Graph | Dominant ops | EP roll-up |
|---|---|---|---|
| fp32 | 1923 ops / 37 types | Add 342; Reshape 252; MatMul 199; Concat 156; Mul 156; Unsqueeze 125; Transpose 121; ReduceMean 104; Gather 101; Pow 58 | QNN: 15 supported types, 0 partial, 0 unsupported, 33 unknown. OpenVINO: 11 supported types, 0 partial, 6 unsupported, 33 unknown. VitisAI: 37 unknown. |

### Reproduce commands

```powershell
$OUT = "temp/repro_Xenova_owlvit-base-patch32_fp32"
$SHAPES = "temp/repro_Xenova_owlvit_shape_config.json"

uv run winml --version
git rev-parse origin/main
uv run winml inspect -m Xenova/owlvit-base-patch32 --format json
uv run winml build -m Xenova/owlvit-base-patch32 -o "$OUT/root_baseline" --ep cpu --device cpu --no-analyze --no-optimize --no-compile --rebuild
uv run winml config -m Xenova/owlvit-base-patch32/onnx/model.onnx --ep cpu --device cpu -o "$OUT/config_cpu_fp32.json" --overwrite
uv run winml build -m Xenova/owlvit-base-patch32/onnx/model.onnx -c examples/recipes/Xenova_owlvit-base-patch32/cpu/cpu/zero-shot-object-detection_fp32_config.json -o "$OUT/build" --ep cpu --device cpu --no-analyze --no-compile --rebuild
uv run python -c "import json, pathlib; pathlib.Path('$SHAPES').write_text(json.dumps({'input_ids':[1,16],'pixel_values':[1,3,768,768],'attention_mask':[1,16]}, indent=2))"
uv run python -c "import onnx, collections; m=onnx.load('$OUT/build/model.onnx', load_external_data=False); print(m.ir_version, [(x.domain, x.version) for x in m.opset_import]); print([(i.name, [d.dim_value or d.dim_param for d in i.type.tensor_type.shape.dim], i.type.tensor_type.elem_type) for i in m.graph.input]); print([(o.name, [d.dim_value or d.dim_param for d in o.type.tensor_type.shape.dim], o.type.tensor_type.elem_type) for o in m.graph.output]); print(collections.Counter(n.op_type for n in m.graph.node).most_common(10))"
uv run winml perf -m "$OUT/build/model.onnx" --ep cpu --device cpu --shape-config "$SHAPES" --iterations 3 --warmup 1 --memory --format json -o "$OUT/perf_cpu.json" --overwrite
uv run winml eval --schema --task zero-shot-object-detection
uv run winml analyze --model "$OUT/build/model.onnx" --ep all --output "$OUT/analyze_all.json"
```

### CODE REVIEW

**Comment by DingmaomaoBJTU** [comment]
Seems the recipe is incorrect

**Comment by fuliucansheng** [comment]
Closing per request.
